infra/azure/terraform/capz/capz-monitoring/main.tf

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+variable "resource_group_name" {
+  type = string
+}
+
+variable "location" {
+  type = string
+}
+
+variable "subscription_id" {
+  type = string
+}
+
+# Create the "capz-monitoring" resource group
+resource "azurerm_resource_group" "capz-monitoring" {
+  location = var.location
+  name     = var.resource_group_name
+  tags = {
+    DO-NOT-DELETE     = "contact capz"
+    creationTimestamp = "2024-10-24T00:00:00Z"
```

Comment:
maybe we could use `timestamp()` here (https://developer.hashicorp.com/terraform/language/functions/timestamp). rg-cleanup uses rf3339 so `timestamp()` should work here https://github.com/Azure/rg-cleanup/blob/1f19a91e03d7c78f0ce1634f1e022a6e8c2713d3/main.go#L31